packages/pancake-uikit/src/components/TabMenu/TabMenu.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { cloneElement, Children, ReactElement } from "react";
+import StyledTabMenu from "./StyledTabMenu";
+import { TabMenuProps } from "./types";
+
+const ButtonMenu: React.FC<TabMenuProps> = ({ activeIndex = 0, onItemClick, children }) => {
+  return (
+    <StyledTabMenu>
+      {Children.map(children, (child: ReactElement, index) => {
+        return cloneElement(child, {
+          isActive: activeIndex === index,
+          onClick: onItemClick ? () => onItemClick(index) : undefined,
+          tabMenuLength: children.length,
+        });
+      })}
```

Comment:
If you want to calculate percentage it seems you can do it here automatically. Is the percentage intended to be consumer controlled?

Reply:
Assuming I need the % (which as you've pointed out, I probably don't) then yea this would be a good amendment!